pkg/reconciler/mongodb/resources/receive_adapter_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/googleinterns/knative-source-mongodb/pkg/apis/sources/v1alpha1"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/kmeta"
+	_ "knative.dev/pkg/metrics/testing"
+	_ "knative.dev/pkg/system/testing"
+)
+
+func TestMakeReceiveAdapter(t *testing.T) {
+	name := "source-name"
+	src := &v1alpha1.MongoDbSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "source-namespace",
+			UID:       "sampleUID",
+		},
+		Spec: v1alpha1.MongoDbSourceSpec{
+			ServiceAccountName: "source-svc-acct",
+			Secret:             corev1.LocalObjectReference{Name: "my-mongo-secret"},
+			Database:           "db",
+			Collection:         "coll",
+		},
+	}
+
+	got, err := MakeReceiveAdapter(&ReceiveAdapterArgs{
+		Image:  "test-image",
+		Source: src,
+		Labels: map[string]string{
+			"test-key1": "test-value1",
+			"test-key2": "test-value2",
+		},
+		SinkURL:        "sink-uri",
+		CeSourcePrefix: "mongodb://",
+	})
+	if err != nil {
+		t.Errorf("Couldn't make Receive Adapter %w", err)
+	}
+	if got != nil {
```

Comment:
remove this if

Reply:
Oops I was using that to debug the null pointer error haha, will remove